lib/galaxy/jobs/__init__.py

```diff
@@ -2092,6 +2092,7 @@ def _change_ownership(self, username, gid):
             log.debug('(%s) Changing ownership of working directory with: %s' % (job.id, ' '.join(cmd)))
             p = subprocess.Popen(cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             stdout, stderr = p.communicate()
+            stdout, stderr = unicodify(stdout).strip(), unicodify(stdout).strip()
```

Comment:
Sorry for being late on this, but this should have been:
```python
            stdout, stderr = unicodify(stdout).strip(), unicodify(stderr).strip()
```

Reply:
That's also a textbook example of why we shouldn't copy-paste our code around 😆

Reply:
Indeed :) Will open a PR with the necessay changes including the other suggestions of @nsoranzo